modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryRelatedObjectsResourceTest.java

```diff
@@ -147,6 +149,229 @@ public void tearDown() throws Exception {
 			_objectDefinition3);
 	}
 
+	@Test
+	public void testCustomObjectNestedFieldOneToManyRelationshipPriorities()
+		throws Exception {
+
+		ObjectRelationship objectRelationship = _addObjectRelationship(
+			_objectDefinition1, _objectDefinition2,
+			ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
+
+		ObjectField objectField = _objectFieldLocalService.getObjectField(
+			objectRelationship.getObjectFieldId2());
+
+		// nestedFields before ID
+
+		JSONAssert.assertEquals(
+			JSONUtil.put(
+				_OBJECT_FIELD_NAME_2, _OBJECT_FIELD_VALUE_2
+			).put(
+				objectRelationship.getName(),
+				JSONUtil.put(
+					_OBJECT_FIELD_NAME_1, _OBJECT_FIELD_VALUE_1
+				).put(
+					"externalReferenceCode",
+					_objectEntry1.getExternalReferenceCode()
+				)
+			).toString(),
+			HTTPTestUtil.invokeToJSONObject(
+				JSONUtil.put(
+					ObjectFieldSettingUtil.getValue(
+						ObjectFieldSettingConstants.
+							NAME_OBJECT_RELATIONSHIP_ERC_OBJECT_FIELD_NAME,
+						objectField),
+					_objectEntry1.getExternalReferenceCode()
+				).put(
+					objectRelationship.getName(),
+					JSONUtil.put(
+						_OBJECT_FIELD_NAME_1, _OBJECT_FIELD_VALUE_1
+					).put(
+						"externalReferenceCode",
+						_objectEntry1.getExternalReferenceCode()
+					).put(
+						"status",
+						JSONUtil.put(
+							"code", 0
+						).put(
+							"label", "approved"
+						).put(
+							"label_i18n", "Approved"
+						)
+					)
+				).put(
+					StringBundler.concat(
+						"r_", objectRelationship.getName(), "_",
+						_objectDefinition1.getPKObjectFieldName()),
+					Long.MAX_VALUE
+				).put(
+					_OBJECT_FIELD_NAME_2, _OBJECT_FIELD_VALUE_2
+				).put(
+					"status",
+					JSONUtil.put(
+						"code", 0
+					).put(
+						"label", "approved"
+					).put(
+						"label_i18n", "Approved"
+					)
+				).toString(),
+				_objectDefinition2.getRESTContextPath(
+				).substring(
+					1
+				),
+				Http.Method.POST
+			).toString(),
+			JSONCompareMode.LENIENT);
+
+		//  ID before invalid ERC
+
+		JSONAssert.assertEquals(
+			JSONUtil.put(
+				_OBJECT_FIELD_NAME_2, _OBJECT_FIELD_VALUE_2
+			).put(
+				StringBundler.concat(
+					"r_", objectRelationship.getName(), "_",
+					_objectDefinition1.getPKObjectFieldName()),
+				(Long)_objectEntry1.getObjectEntryId()
```

Comment:
Why are you casting the response? I think it already returns a `long`

Reply:
The assert equal the long object differently the primitive long fails with:

java.lang.AssertionError: r_qhgs_c_a73j8IOjMId
    Expected: 33557
    got: 33557